File: web/src/app/user/models/user-unit.settings.ts
import { SpeedUnit } from '@core/models/api/response/speed.unit';
import { TemperatureUnit } from '@core/models/api/response/temperature.unit';
import { HeightUnit } from '@core/models/api/response/height.unit';
import { PercentageUnit } from '@core/models/api/response/percentage.unit';
import { DirectionUnit } from '@core/models/api/response/direction.unit';
import { PressureUnit } from '@core/models/api/response/pressure.unit';

export type UserUnitSettings = {
  speed: SpeedUnit;
  temperature: TemperatureUnit;
  height: HeightUnit;
  percentage: PercentageUnit;
  direction: DirectionUnit;
  pressure: PressureUnit;
};

File: backend/src/main/java/com/weather/backend/user/models/UserUnitSettings.java
package com.weather.backend.user.models;

import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@Getter
@Setter
@Builder
public class UserUnitSettings {
    private String speed;
    private String temperature;
    private String height;
    private String percentage;
    private String direction;
    private String pressure;

    public UserUnitSettings(String speed, String temperature, String height, String percentage, String direction, String pressure) {
        this.speed = speed;
        this.temperature = temperature;
        this.height = height;
        this.percentage = percentage;
        this.direction = direction;
        this.pressure = pressure;
    }
}
